src/view.rs

```diff
@@ -37,101 +36,109 @@ pub fn render_game(app: &mut ApplicationState, ctx: &mut Context) -> GameResult<
-        let height = if is_title { h - drawable.height() as f32 } else { h };
-        let point = Point2::new(width, height);
-        graphics::set_color(ctx, color).unwrap();
-        graphics::draw(ctx, &drawable, point, 0.0).unwrap();
-    };
-
-    draw_text(title, color::WHITE, true);
-    draw_text(subtitle, color::GREY, false);
+fn render_message(ctx: &mut Context, app: &mut ApplicationState) -> GameResult<()> {
+    if let Some(ref message) = app.game_state.message {
+        let Message { title, subtitle } = *message;
+        let Size { width, height } = app.game_state.world.size;
+
+        let w = width as f32 / 2.0;
+        let h = height as f32 / 2.0;
+
+        let mut draw_text = |text: &str, color: graphics::Color, is_title: bool| {
+            let drawable = graphics::Text::new(ctx, text, &app.resources.font).unwrap();
+            let width = w - (drawable.width() as f32 / 2.0);
+            let height = if is_title { h - drawable.height() as f32 } else { h };
+            let point = Point2::new(width, height);
+            graphics::set_color(ctx, color).unwrap();
+            graphics::draw(ctx, &drawable, point, 0.0).unwrap();
+        };
+
+        draw_text(title, color::WHITE, true);
+        draw_text(subtitle, color::GREY, false);
+    }
 
     Ok(())
 }
 
 /// Renders the world and everything in it
-pub fn render_world(world: &World, ctx: &mut Context) -> GameResult<()> {
+pub fn render_world(ctx: &mut Context, world: &World, resources: &Resources) -> GameResult<()> {
     // Render stars in the background
     graphics::set_color(ctx, color::GREY)?;
-    render_stars(world, ctx)?;
+    render_stars(ctx, world, resources)?;
 
     // Draws particles in violet
-    graphics::set_color(ctx, color::VIOLET)?;
-    for particle in &world.particles {
-        render_particle(particle, ctx)?;
-    }
+    graphics::set_color(ctx, color::ORANGE)?;
+    render_particles(ctx, world, resources)?;
     
     // Draw any bullets as blue
     graphics::set_color(ctx, color::BLUE)?;
     for bullet in &world.bullets {
-        render_bullet(bullet, ctx)?;
+        render_bullet(ctx, bullet)?;
     }
 
     // Now we draw the enemies as yellow
     graphics::set_color(ctx, color::YELLOW)?;
     for enemy in &world.enemies {
-        render_enemy(enemy, ctx)?;
+        render_enemy(ctx, enemy)?;
     }
 
     // Finally draw the player as red
     if !world.player.is_dead {
         graphics::set_color(ctx, color::RED)?;
-        render_player(&world.player, ctx)?;
+        render_player(ctx, &world.player)?;
     }
 
     Ok(())
 }
 
 /// Renders all the stars in the background
-fn render_stars(world: &World, ctx: &mut Context) -> GameResult<()> {
-    let img = graphics::Image::solid(ctx, 1, color::GREY)?;
-    let mut sb = SpriteBatch::new(img);
+fn render_stars(ctx: &mut Context, world: &World, resources: &Resources) -> GameResult<()> {
+    let mut spritebatch = SpriteBatch::new(resources.star_image.clone());
```

Comment:
Cloning the image on each frame seems like a waste. It probably doesn't matter that much for this game because the images are small, but still... Do you think we could reuse the `SpriteBatch` accross drawings (for instance, by treating it as a resource)? This seems the cleanest option to me, but I don't know whether `SpriteBatch` is designed for this use case. It seems worth trying, though.